### PR TITLE
Mark python 2.7 as dead

### PR DESCRIPTION
**Pull request by RDIL**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@RDIL

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by benjaminp** [comment]
This seems fine, though it'd be nice to have the CLA signed.

**Comment by RDIL** [comment]
Yeah, I actually signed it yesterday.

**Comment by RDIL** [comment]
My guess is that the secretary hasn't filed the CLA yet in the records.

File: index.rst [hunk 1]
```diff
@@ -99,7 +99,7 @@ Status of Python branches
 +------------------+--------------+-------------+----------------+----------------+-------------------+
 | 3.7              | :pep:`537`   | bugfix      | 2018-06-27     | *2023-06-27*   | Ned Deily         |
 +------------------+--------------+-------------+----------------+----------------+-------------------+
-| 2.7              | :pep:`373`   | bugfix      | 2010-07-03     | *2020-01-01*   | Benjamin Peterson |
+| 2.7              | :pep:`373`   | unsupported | 2010-07-03     | *2020-01-01*   | Benjamin Peterson |
```
Comment (terryjreedy) [inline index.rst]: This is a table of active branches.  Normally, a branch keeps its (security) status until marked EOL and moved to the EOL table in https://devguide.python.org/devcycle/#end-of-life-branches, after final release. 

2.7 continues to need special handling.  It is not getting a formal security-fix-only stage and is instead getting a few months of 'near EOL'.  With the change in status to 'unsupported', it should be moved out of the 'bugfix' group to the bottom of this table.  'unsupported' should be temporarily added to the 'Status' table below this one (until the above happens), with an entry such as
`   release cycle is frozen except as needed for final release.`

File: index.rst [hunk 2]
```diff
@@ -127,8 +127,8 @@ Dates in *italic* are scheduled and can be adjusted.
 
 By default, the end-of-life is scheduled 5 years after the first release,
 but can be adjusted by the release manager of each branch.  The support for
-Python 2.7 has currently been extended to 2020-01-01.  Versions older than
-2.7 have reached end-of-life.
+Python 2.7 was extended to 2020-01-01.  All Python 2 versions have reached
+end-of-life.
```
Comment (terryjreedy) [inline index.rst]: Once 2.7 in moved to complete EOL, the 2.7-specific extension sentence should be removed.  But I think the Py 2 EOL sentence should be left.

File: index.rst (hunk 1 above)
Comment (zware) [reply index.rst]: I'd suggest to just go ahead and remove 2.7 here entirely.  It is already EOL from a development perspective, and removing it now means we don't have to remember to remove it later :)

**Comment by jdufresne** [comment]
I think an entry should be added to the "End of life" section:

https://github.com/python/devguide/blob/master/devcycle.rst#end-of-life-branches

**Comment by RDIL** [comment]
Done @jdufresne @terryjreedy

**Pull request merged**